arch/arm/core/aarch32/mpu/arm_mpu_v7_internal.h

```diff
@@ -172,6 +172,8 @@ static inline int mpu_buffer_validate(void *addr, size_t size, int write)
 {
 	int32_t r_index;
 
+	int key = arch_irq_lock();
```

Comment:
yeah this change looks necessary.
I wonder if this issue this fixes is reproducible in the NXP or ARMv8 version of this MPU code.